### DISCUSSION TITLE
Add parameter "reason" to app.bsky.notification.listNotifications

### DISCUSSION THREAD

**Discussion by seboslaw** [bluesky-social/atproto#1691] [Bluesky Client App]
Hey guys,

I was wondering if it's possible to add the optional parameter "reason" to app.bsky.notification.listNotifications? That way we can filter what notifications we want to load. In my app I'd like to add a segmented control so that users can choose whether to display all notification or only mentions. 
Filtering on the client doesn't work since the request is batched and the first batch may very well not have any mentions in it.

**Comment by bnewbold** [comment]
Just to clarify, you mean add `reason` as a query parameter when requesting the endpoint, and have the server return only notifications matching that single reason, is that right?

I believe the way most clients have been handling this is to filter the response. For the "number of unread notifications" endpoint, that number is used mostly as an indication that there *might* be unread notifications, and the client paginates through with the cursor to count how many relevant notifications there actually are.

**Reply by seboslaw** [reply]
Yes, exactly: add `reason` as a query parameter. 
Filtering the responses on the client-side doesn't really work with pagination. Let's say the page-limit is set to 50 my first 200 notifications are only likes and quotes. I would have to do 5 request until I would get my first "mention" to display. 
In general, this could result in doing endless pagination requests until I find the post I want to display.

**Reply by bnewbold** [reply]
cc: @pfrazee @dholms this seems like a reasonable request, you might have already thought about this.

From a prioritization standpoint, it is *possible* to just page through (many requests) to do the behavior client-side, this would be an efficiency and dev quality of life thing.

**Reply by pfrazee** [reply]
It is reasonable, and likely necessary for what is being attempted. I think you'll likely want to be able to specify multiple reasons in the query parameter to do the intended behavior, or else you'll need to send multiple requests in parallel and merge them together which is a chore.

Ultimately this is a question of which indexes the server can maintain and what we want to enshrine in the protocol as a hard requirement. This might require formalizing the reason field a bit, or developing the API a bit to encompass the concepts in a forward-looking way.

**Reply by seboslaw** [reply]
Hey @bnewbold, I was wondering what the status of this issue is? Will this be implemented?

**Reply by bnewbold** [reply]
@seboslaw following up on this, we are very focused on getting to open federation, scaling, and finishing some moderation features. Adding this parameter seems like a good improvement, and we will almost certainly get around to it, but can't I can't give you any specific or ballpark timeline right now, sorry.
